lib/Test2/Compare/Negatable.pm

```diff
@@ -0,0 +1,119 @@
+package Test2::Compare::Negatable;
+use strict;
+use warnings;
+
+require overload;
+require Test2::Util::HashBase;
+
+sub import {
+    my ($pkg, $file, $line) = caller;
+
+    my $sub = eval <<"    EOT" or die $@;
+package $pkg;
+#line $line "$file"
+sub { overload->import('!' => 'clone_negate', fallback => 1); Test2::Util::HashBase->import('negate')}
```

Comment:
> It is necessary because compare objects are reusable.

What I was asking about was why the overload was applied in such a contorted manner. A normal operator overload can clone if it wants. I understand now that it's because it's a role.